### PR TITLE
implement drag-and-drop feature using react-beautiful-dnd so users can drag their items to containers

### PR DESCRIPTION
**Pull request by Jeffchiucp**
* install react drag-n-drop library(https://github.com/atlassian/react-beautiful-dnd)

* update react-redux states for drag-n-drop

* add redux states for droppable and incorporate the drag-n-drop library. A Droppable list can be a scroll container (without a scrollable parent) or be the child of a scroll container (that also does not have a scrollable parent)

* add the function getItemStyle to the utils folder.  This function add some basic styles to make the items look a bit nicer, change background color if dragging.

* Add Auto scrolling
When a user drags a Draggable near the edge of a container we automatically scroll the container as we are able to in order make room for the Draggable.

### CODE REVIEW

File: src/components/item/index.js [hunk 1]
```diff
@@ -1,44 +1,65 @@
...
+    const options = [
+      'Delete'
+    ]
+    
+    return (
+      <Draggable
+        key={this.props.item._id}
+        draggableId={this.props.item._id}
+        index={this.props.index}
+      >
```
Comment (deonjo) [inline src/components/item/index.js]: please place this on end of previous line, and all other places.

File: src/views/arrange/index.js [hunk 2]
```diff
@@ -117,39 +121,104 @@ export class Arrange extends Component {
     ) */
   }
 
+  onDragEnd = result => {
+    const { source, destination } = result
+    if (!destination) {
+      return
+    }
+    
+    if (source.droppableId === destination.droppableId) {
```
Comment (deonjo) [inline src/views/arrange/index.js]: can you comment what happens in each of these if else block?

Comment (Jeffchiucp) [reply src/views/arrange/index.js]: sure.

File: src/components/item/index.js (hunk 1 above)
Comment (Jeffchiucp) [reply src/components/item/index.js]: sure.

File: src/views/arrange/index.js (hunk 2 above)
Comment (Jeffchiucp) [reply src/views/arrange/index.js]: Yes, I have added the comments to each if else block.

**Pull request merged**